lib/facter/custom_facts/core/execution/base.rb

```diff
@@ -77,12 +79,31 @@ def builtin_command?(command)
           output.chomp =~ /builtin/ ? true : false
         end
 
-        def execute_command(command, on_fail, logger = nil)
+        def execute_command(command, on_fail, logger = nil, time_limit = nil)
+          @log ||= Log.new(self)
+          time_limit ||= 1.5
           begin
             # Set LC_ALL and LANG to force i18n to C for the duration of this exec;
             # this ensures that any code that parses the
             # output of the command can expect it to be in a consistent / predictable format / locale
-            out, stderr, _status_ = Open3.capture3({ 'LC_ALL' => 'C', 'LANG' => 'C' }, command.to_s)
+            opts = { 'LC_ALL' => 'C', 'LANG' => 'C' }
+            require 'timeout'
+            @log.debug("Executing command: #{command}")
```

Comment:
Why do you use 2 loggers? One initialised in this method and one received as parameter

Reply:
There are 2 faulty scenarios that we want to be logged different: 
1st when the command timeout because it takes more than the time limit to execute (this will be logged with Facter::Core::Execution specific logger)
2nd the command doesn't exceed the time limit but the command still fails and it produces some output on stderr (this will be logged with the logger from the specific resolver that the command came from )